utils/common/src/main/java/org/apache/brooklyn/util/JavaGroovyEquivalents.java

```diff
@@ -66,8 +66,9 @@ public static String elvis(String preferred, String fallback) {
     public static String elvisString(Object preferred, Object fallback) {
         return elvis(asString(preferred), asString(fallback));
     }
+    @SuppressWarnings("unchecked")
     public static <T> T elvis(T preferred, T fallback) {
-        return groovyTruth(preferred) ? preferred : fallback;
+        return (T) fix(groovyTruth(preferred) ? preferred : fallback);
```

Comment:
Having `fix` is concerning I'd rather drop GString that may have been supplied.
Also having different behavior is concerning.
```java
    public static String elvis(String preferred, String fallback) {
        return groovyTruth(preferred) ? preferred : fallback;
    }
//..
// and a fixed version
//..
    public static <T> T elvis(T preferred, T fallback) {
        return (T) fix(groovyTruth(preferred) ? preferred : fallback);
    }
```